### ISSUE TITLE
TabSheet internal tabs customization

### ISSUE THREAD

**Issue by gschrader** [vaadin/flow-components#5232]
### Describe your motivation

I wanted to be able to set the tabs orientation to vertical.

### Describe the solution you'd like

maybe add a method to allow to customize the tabs instance in case further methods become available:

In the class TabSheet:
```
    public void customize(Consumer<Tabs> tabsConsumer) {
        tabsConsumer.accept(tabs);
    }
```

my usage to set vertical orientation would be 

```
            tabSheet.customize(tabs -> {
                tabs.setOrientation(Tabs.Orientation.VERTICAL);
            });

```

### Describe alternatives you've considered

I'm using reflection for now 🤮: 
```
            Field field = TabSheet.class.getDeclaredField("tabs");
            field.setAccessible(true);
            Tabs tabs = (Tabs) field.get(springTabSheet);
            tabs.setOrientation(Tabs.Orientation.VERTICAL);
```

**Comment by yuriy-fix** [comment]
We could consider exposing `Tabs` directly as a getter.

**Comment by gschrader** [comment]
Yeah a getter would work, I'm not sure why I didn't just propose that.

**Comment by TatuLund** [comment]
There is an alternative TabSheet component in Directory which has a feature of vertical Tabs https://vaadin.com/directory/component/tabsheet

**Comment by gschrader** [comment]
I would prefer to stick with the official component for easier upgrades down the road.

**Comment by yuriy-fix** [comment]
Additional context: https://github.com/vaadin/flow-components/issues/5370

**Comment by tomivirkki** [comment]
I'm not sure if exposing `Tabs` in `TabSheet` is a good idea. Giving the user full control over the internal component may lead to unexpected results.

For example, with the following `TabSheet`:

![Screenshot 2023-08-31 at 12 03 02](https://github.com/vaadin/flow-components/assets/1222264/b04c0feb-a426-412a-8048-73532b818c94)

```java
var tabsheet = new TabSheet();
tabsheet.setPrefixComponent(new Span("Prefix"));
tabsheet.setSuffixComponent(new Span("Suffix"));
tabsheet.add("Tab one", new Span("Tab one content"));
tabsheet.add("Tab two", new Span("Tab two content"));
add(tabsheet);
```

If you try to make the Tabs vertical with `tabsheet.getTabs().setOrientation(Orientation.VERTICAL);` (as implemented by https://github.com/vaadin/flow-components/pull/5408) you'd end up with:

![Screenshot 2023-08-31 at 12 03 17](https://github.com/vaadin/flow-components/assets/1222264/a38e0f8f-ec26-4fa1-8ecc-87fefdbf6eab)

...which is probably not what you wanted here. If you want a vertical `TabSheet`, you probably want it to look more like:

<img width="422" alt="Screenshot 2023-08-31 at 12 08 33" src="https://github.com/vaadin/flow-components/assets/1222264/3ae672d1-149c-49e2-9c42-dc4ea0b719a1">

**Comment by gschrader** [comment]
@tomivirkki you're right, I knew something didn't look quite right but it took your screenshots for me to realize. I have since gone with a different layout so I'm not sure what you want to do with this open issue.

**Comment by Avec112** [comment]
I also would like to see `TabSheet` with vertical orientation. Like gschrader I also do not want to install components not part of Vaadin Flow.

**Comment by knoobie** [comment]
> ...which is probably not what you wanted here.

Easy workaround:

```css

vaadin-tabsheet {
  flex-direction: row;
}
```

**Comment by fredpena** [comment]
Thas allows me to set up the `TabSheet` vertically.
```java
        final TabSheet tabSheet = new TabSheet();
        tabSheet.addClassNames(LumoUtility.Display.FLEX, LumoUtility.FlexDirection.ROW);

        Tabs tabs = (Tabs) tabSheet.getChildren()
                .filter(c -> c.getClass().equals(Tabs.class))
                .findFirst().orElse(null);

        if (tabs != null) {
            tabs.getElement().setAttribute("orientation", "vertical");
        }
```

**Comment by geertschuring** [comment]
I got confused about the use of Tabs and Tabsheet. Feels like an unfinished development. I need Tabsheet to bind a tab and some content but need Tabs to get vertical orientation but then need to figure out how to show the correct content when a tab is clicked. Not very difficult but I don't see how someone looks at this situation and concludes that it makes sense.

Please just add the setOrientation method to the TabSheet component.

**Comment by geertschuring** [comment]
> > ...which is probably not what you wanted here.
> 
> Easy workaround:
> 
> vaadin-tabsheet {
>   flex-direction: row;
> }

This does not work for me. The tabs are still horizontal but the content is shown vertically to the complete set of tabs.

**Comment by geertschuring** [comment]
> We could consider exposing `Tabs` directly as a getter.

I don't think this is a good idea. As a developer I just want to create a bunch of Tabs, add them to a TabSheet, set its orientation and be done. I would consider Tabs to be a internal structure of TabSheet and never touch it.
